src/components/chip/chip.tsx

```diff
@@ -0,0 +1,289 @@
+import { getIconName } from '../icon/get-icon-props';
+import {
+    makeEnterClickable,
+    removeEnterClickable,
+} from '../../util/make-enter-clickable';
+import translate from '../../global/translations';
+import {
+    BACKSPACE,
+    BACKSPACE_KEY_CODE,
+    DELETE,
+    DELETE_KEY_CODE,
+} from '../../util/keycodes';
+
+/**
+ * Chips and buttons are both interactive elements in UI design,
+ * but they serve different purposes and are used in different contexts.
+ *
+ * :::warning
+ * Do not use the chip component carelessly, as an alternative for
+ * [`limel-button`](#/component/limel-button/) in the UI design!
+ *
+ * **Buttons:**
+ * Buttons are used to trigger actions. They are typically used to
+ * submit forms, open dialogs, initiate a process, or perform any action
+ * that changes the state of the application.
+ * Buttons' labels usually contain action words, in other words, the labels is
+ * a _verb in imperative mood_ such as "Submit" or "Delete".
+ * Buttons are placed in areas where it's clear they will initiate
+ * an action when clicked.
+ *
+ * **Chips:**
+ * Chips however are elements which may look like buttons, but they are
+ * representing choices, filters, or tags, in a small block
+ * or clearly bundled into a group. Chips are rarely used alone in the
+ * user interface.
+ * They are often used in a so called "chip-set", or placed together in
+ * a section of the UI, where the user can expect more than one chip to be present.
+ *
+ * For example, a chip may represent a filter in a filter bar, or a tag in a tag list,
+ * or an item in a shopping list.
+ * Clicking a chip can also trigger an action, for example toggling a filter ON or OFF,
+ * or opening a page with all posts tagged with the tag represented by the chip,
+ * or navigating to a page with more information about the item in the shopping list.
+ * :::
+ *
+ * @private
+ * @exampleComponent limel-example-chip-button
+ * @exampleComponent limel-example-chip-link
+ * @exampleComponent limel-example-chip-icon-colors
+ * @exampleComponent limel-example-chip-badge
+ * @exampleComponent limel-example-chip-filter
+ * @exampleComponent limel-example-chip-removable
+ * @exampleComponent limel-example-chip-aria-role
+ */
+@Component({
+    tag: 'limel-chip',
+    shadow: true,
+    styleUrl: 'chip.scss',
+})
+export class Chip {
+    /**
+     * Defines the language for translations.
+     * Will translate the translatable strings on the components.
+     */
+    @Prop()
+    public language: Languages = 'en';
+
+    /**
+     * ID of the chip. Must be unique.
+     */
+    @Prop()
+    id: number | string;
+
+    /**
+     * Label displayed on the chip
+     */
+    @Prop()
+    public text: string;
+
+    /**
+     * Icon of the chip.
+     */
+    @Prop()
+    public icon: string | Icon;
+
+    /**
+     * If supplied, the chip will become a clickable link.
+     */
+    @Prop()
+    public link?: Omit<Link, 'text'>;
```

Comment:
No, I didn't forget these 😊 These should not be reflected, since they are not primitive values. (Well, `icon` _might_ be primitive, but also might not be…)

Reply:
OK I don't really know what the "primitive values" are. But in my world, if the reflected value can be used for styling, I reflect it. Otherwise, I keep it quite 😅 is that not enough? 

What I've learned as a general rule of thumb, you should only use `@Prop({ reflect: true })` for props that are used to control the visual appearance of the component. For all other props, you should use `@Prop()`. This will make your code more maintainable and easier to understand.

I'm thinking, knowing the `text` or the `badge` or `icon` or even the `id` has no value in styling the component. So why reflecting them?

Reply:
> I don't really know what the "primitive values" are.

Basically it's strings, numbers, and booleans. Technically `null` and `undefined` are also primitive values, and I assume `NaN` is too, since it's considered a `number` (yes! "not a number" has the type `number` in javascript 😂)

(Technically, anything that isn't a function or an object is a primitive, and since functions are also objects, anything that isn't an object is a primitive 😝)